### ISSUE TITLE
@MicronautTest doesn't have support for starting AWS lambda functions that extend the 'MicronautRequestHandler' class

### ISSUE THREAD

**Issue by athkalia** [micronaut-projects/micronaut-test#330]
It seems like there is no easy way to instantiate a lambda function by using the `@MicronautTest` annotation.

In the AWS lambda function tutorials, e.g. here https://guides.micronaut.io/mn-serverless-function-aws-lambda-kotlin/guide/index.html it suggested that to test a handler you just instantiate it. That, however, makes it difficult to start a handler with a specific environment or other configuration that is typically passed via the `@MicronautTest` annotation.

If a `@MicronautTest` is defined and the handler is injected instead of created manually, we still get Micronaut starting up twice, as the constructor of the `MicronautRequestHanlder` looks like this:

```
public MicronautRequestHandler() {
        buildApplicationContext(null);
        applicationContext.inject(this);
    }
```
So the handler instantiation creates a new standalone `ApplicationContext`

The only way I could make this work was to not use the `@MicronautTest`annotation, and instead pass a system property before running the test, like:

```
System.setProperty("micronaut.environments", "myenv")
val handler = MyHandlerRepoCheckoutHandler()
```

But that seems a bit odd.

As a side-note, when I set the environment manually before starting the handler, as soon as the handler started I got an error:

```
Bean definition [io.micronaut.test.transaction.TestTransactionInterceptorListener] could not be loaded: Failed to inject value for parameter [rollback] of class: io.micronaut.test.transaction.TestTransactionInterceptorListener

Message: Error resolving property value [micronaut.test.rollback]. Property doesn't exist
Path Taken: new TestTransactionInterceptorListener(TestTransactionInterceptor interceptor,[boolean rollback])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.test.transaction.TestTransactionInterceptorListener] could not be loaded: Failed to inject value for parameter [rollback] of class: io.micronaut.test.transaction.TestTransactionInterceptorListener

Message: Error resolving property value [micronaut.test.rollback]. Property doesn't exist
Path Taken: new TestTransactionInterceptorListener(TestTransactionInterceptor interceptor,[boolean rollback])
    at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1549)
    at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:220)
    at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2780)
    at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:233)
    at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:166)
    at io.micronaut.function.executor.AbstractExecutor.startEnvironment(AbstractExecutor.java:125)
    at io.micronaut.function.aws.MicronautRequestHandler.buildApplicationContext(MicronautRequestHandler.java:85)
    at io.micronaut.function.aws.MicronautRequestHandler.<init>(MicronautRequestHandler.java:48)
    at com.caribou.codeimport.repocheckout.RepoCheckoutHandler.<init>(RepoCheckoutHandler.kt:17)
    at com.caribou.codeimport.repocheckout.RepoCheckoutHandlerTest.handler passes correct details to the checkout service$code_import(RepoCheckoutHandlerTest.kt:26)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```

Which I solved by adding a property in my `application-myenv.yml` file:
```
micronaut:
  test:
    rollback: false
```

**Comment by graemerocher** [comment]
contributions welcome!

**Comment by graemerocher** [comment]
Fixed by #336
